pom.xml

```diff
@@ -389,6 +389,24 @@
                 <show>protected</show>
                 <nohelp>true</nohelp>
                 <outputDirectory>${project.build.directory}/javadoc</outputDirectory>
+                <groups>
+                  <group>
+                    <title>Main packages</title>
+                    <packages>com.google.gcloud*</packages>
+                  </group>
+                  <group>
+                    <title>Test helpers packages</title>
+                    <packages>com.google.gcloud.bigquery.testing:com.google.gcloud.datastore.testing:com.google.gcloud.resourcemanager.testing:com.google.gcloud.storage.testing</packages>
+                  </group>
+                  <group>
+                    <title>SPI packages</title>
+                    <packages>com.google.gcloud.spi</packages>
```

Comment:
+1 for s/Main packages/API packages/

My preference would be to order based on usefulness to users and therefore I would put testing before SPI (and maybe even consider examples before SPI). But not a strong opinion.